src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -670,173 +637,26 @@ export const handleEncryptedSubmission = [
-    logger.warn({
-      message: 'Failed to retrieve form from database',
-      meta: logMeta,
-      error: formResult.error,
-    })
-    const { errorMessage, statusCode } = mapRouteError(formResult.error)
-    return res.status(statusCode).json({ message: errorMessage })
-  }
-
-  // Guardrail to prevent new endpoint from being used for regular storage mode forms.
-  // TODO (FRM-1232): remove this guardrail when encryption boundary is shifted.
-  if (!formResult.value.get('newEncryptionBoundary')) {
-    return res
-      .status(StatusCodes.FORBIDDEN)
-      .json({ message: 'This endpoint has not been enabled for this form.' })
-  }
-
-  // Retrieve public key.
-  const publicKey = formResult.value.publicKey
-
-  if (!publicKey) {
-    logger.warn({
-      message: 'Form does not have a public key',
-      meta: logMeta,
-    })
-    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
-      message: 'Form does not have a public key',
-    })
-  }
-
-  const attachmentsMap: Record<string, Buffer> = {}
-
-  // Populate attachment map
-  req.body.responses.filter(isAttachmentResponse).forEach((response) => {
-    const fieldId = response._id
-    attachmentsMap[fieldId] = response.content
-  })
-
-  const encryptedAttachments =
-    await getEncryptedAttachmentsMapFromAttachmentsMap(
-      attachmentsMap,
-      publicKey,
-    )
-
-  const filteredResponses = getFilteredResponses(
-    formResult.value,
-    req.body.responses,
-  )
-
-  if (filteredResponses.isErr()) {
-    logger.warn({
-      message: filteredResponses.error.message,
-      meta: logMeta,
-    })
-    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
-      message: filteredResponses.error.message,
-    })
-  }
-
-  const encryptedContent = formsgSdk.crypto.encrypt(
-    req.body.responses.map((response) => {
-      if (isAttachmentResponse(response)) {
-        return {
-          ...response,
-          filename: undefined,
-          content: undefined, //Strip out attachment content
-        }
-      } else {
-        return response
-      }
-    }),
-    publicKey,
-  )
-
-  const encryptedVersion = {
-    attachments: encryptedAttachments,
-    responses: filteredResponses.value as EncryptFormFieldResponse[],
-    encryptedContent,
-    version: req.body.version,
-  }
-
-  req.body = encryptedVersion
-  return next()
-}
-
 export const handleStorageSubmission = [
   CaptchaMiddleware.validateCaptchaParams,
   TurnstileMiddleware.validateTurnstileParams,
   ReceiverMiddleware.receiveStorageSubmission,
   EncryptSubmissionMiddleware.validateStorageSubmissionParams,
-  encryptSubmission,
+  EncryptSubmissionMiddleware.createFormsgReqBody,
+  EncryptSubmissionMiddleware.retrieveForm,
+  EncryptSubmissionMiddleware.checkNewBoundaryEnabled,
+  EncryptSubmissionMiddleware.checkPublicKey,
+  EncryptSubmissionMiddleware.validateSubmission,
+  EncryptSubmissionMiddleware.encryptSubmission,
+  EncryptSubmissionMiddleware.checkEncryptMode,
```

Comment:
likewise here, can consider grouping so its neater

Reply:
Addressed in [e081264](https://github.com/opengovsg/FormSG/pull/6595/commits/e081264c64aa0ebb92069bba5f1a7e8c5461c91d), closing to streamline discussions [here](https://github.com/opengovsg/FormSG/pull/6595#discussion_r1298348792).